halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/SpinnakerRuntimeSettings.java

```diff
@@ -36,94 +35,73 @@
   // For serialization
   public SpinnakerRuntimeSettings() {}
 
-  @Data
-  public class Services {
-    ServiceSettings clouddriver;
-    ServiceSettings clouddriverBootstrap;
-    ServiceSettings consulClient;
-    ServiceSettings consulServer;
-    ServiceSettings deck;
-    ServiceSettings echo;
-    ServiceSettings fiat;
-    ServiceSettings front50;
-    ServiceSettings gate;
-    ServiceSettings igor;
-    ServiceSettings kayenta;
-    ServiceSettings orca;
-    ServiceSettings orcaBootstrap;
-    ServiceSettings rosco;
-    ServiceSettings redis;
-    ServiceSettings redisBootstrap;
-    ServiceSettings monitoringDaemon;
-    ServiceSettings vaultClient;
-    ServiceSettings vaultServer;
+  public static class Services extends HashMap<Type,ServiceSettings> {
+    @Override
+    public ServiceSettings get(Object key) {
+      if (!Type.class.isInstance(key)) {
+        throw new HalException(Severity.FATAL, "Key must be of type SpinnakerService.Type");
+      }
+      if (!this.containsKey(key)) {
+        throw new HalException(Severity.FATAL, "Service " + Type.class.cast(key).getCanonicalName() + " does not exist");
+      }
+      return super.get(key);
+    }
+
+    // Getters for backwards compatibility
+    @JsonIgnore
```

Comment:
Why @JsonIgnore? is this being serialized and sent from the daemon to the client?